components/Buttons/tests/unit/ButtonsThemingTest.swift

```diff
@@ -216,5 +216,134 @@ class ButtonsThemingTest: XCTestCase {
+    // Given
+    let button = MDCFloatingButton()
+    let defaultButton = MDCFloatingButton()
+    let scheme = MDCContainerScheme()
+    let colorScheme = MDCSemanticColorScheme(defaults: .material201804)
+    let typographyScheme = MDCTypographyScheme(defaults: .material201804)
+
+    // When
+    button.applySecondaryTheme(withScheme: scheme)
+
+    // Then
+    XCTAssertEqual(button.backgroundColor(for: .normal), colorScheme.secondaryColor)
+    XCTAssertEqual(button.imageTintColor(for: .normal), colorScheme.onSecondaryColor)
+    // XCTAssertEqual(button.titleColor(for: .normal), colorScheme.onSecondaryColor)
+    XCTAssertEqual(button.shadowColor(for: .normal), defaultButton.shadowColor(for: .normal))
+    XCTAssertEqual(button.inkColor, defaultButton.inkColor)
+
+    XCTAssertEqual(button.titleFont(for: .normal), typographyScheme.button)
+
+    if let buttonShape = button.shapeGenerator as? MDCRectangleShapeGenerator {
+      let corner = MDCCornerTreatment.corner(withRadius: 0.5)
+      corner?.valueType = .percentage
+      XCTAssertEqual(buttonShape.topLeftCorner, corner)
+      XCTAssertEqual(buttonShape.topRightCorner, corner)
+      XCTAssertEqual(buttonShape.bottomRightCorner, corner)
+      XCTAssertEqual(buttonShape.bottomLeftCorner, corner)
+    } else {
+      XCTFail("Button.shapeGenerator was not a MDCRectangularShapeGenerator")
+    }
+
+    // Test remaining properties
+    XCTAssertEqual(button.elevation(for: .normal), ShadowElevation.fabResting)
+    XCTAssertEqual(button.elevation(for: .highlighted), ShadowElevation.fabPressed)
+    XCTAssertEqual(button.elevation(for: .disabled), ShadowElevation.none)
+
+    XCTAssertEqual(button.borderWidth(for: .normal), defaultButton.borderWidth(for: .normal), accuracy: 0.001)
+    XCTAssertEqual(button.borderWidth(for: .selected), defaultButton.borderWidth(for: .selected), accuracy: 0.001)
+    XCTAssertEqual(button.borderWidth(for: .highlighted), defaultButton.borderWidth(for: .highlighted), accuracy: 0.001)
+    XCTAssertEqual(button.borderWidth(for: .disabled), defaultButton.borderWidth(for: .disabled), accuracy: 0.001)
+  }
+
+  func testFloatingButtonSecondaryThemeWithCustomColorTheme() {
+    // Given
+    let button = MDCFloatingButton()
+    let defaultButton = MDCFloatingButton()
+    let scheme = MDCContainerScheme()
+    let customColorScheme = MDCSemanticColorScheme()
+    customColorScheme.secondaryColor = .purple
+    customColorScheme.onSecondaryColor = .purple
+    scheme.colorScheme = customColorScheme
+    let typographyScheme = MDCTypographyScheme(defaults: .material201804)
+
+    // When
+    button.applySecondaryTheme(withScheme: scheme)
+
+    // Then
+    XCTAssertEqual(button.backgroundColor(for: .normal), customColorScheme.secondaryColor)
+    XCTAssertEqual(button.imageTintColor(for: .normal), customColorScheme.onSecondaryColor)
+    // XCTAssertEqual(button.titleColor(for: .normal), colorScheme.onSecondaryColor)
```

Comment:
Please add the TODO here and below as well.

Reply:
Addressed in1b3abf543a18d093735c7636cbe962a3831b11dd